### PR TITLE
Added content

### PR DESCRIPTION
**Pull request by jordanhudgens**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,54 +1,91 @@
...
-3. Draw a route with get, mapping a url to a controller action
-4. Create a new controller by hand and inherit from application controller
-5. Define a controller action by defining a method in a controller
-6. Identify missing template errors from Rails (going to an action with no template) and how to solve it (by creating a template)
-7. Use render to explicity render a template
-8. Define the Reails implicit rendering convention
-9. Define the steps required to create a static request
+How does your application know what view to render to users? This is where routing comes in, as a framework Rails has a comprehensive routing system for both dynamic and static pages. Below are the differences between a static and dynamic route:
 
+* **Static route** - A static route will render a view that does not change, you typically will not send parameters to it, examples would be: a site's about or contact pages.
```
Comment (adamjonas) [inline README.md]: Good examples.

**Comment by jordanhudgens** [comment]
@AnnJohn Added metadata

File: README.md [hunk 2]
```diff
@@ -1,54 +1,91 @@
...
+1. A URL is entered into the browser, this is the HTTP request
+2. That request is sent to the server where the application's router interprets the request and sends a message to the controller mapped to that route
+3. The controller communicates with the view file mapped to the controller method
+4. The server returns that HTTP response, which contains the view and page that can be viewed in the browser
 
-Use the example of how we want to have an 'about' page on our site. So let's try to go there, let's just type localhost:3000/about.
+## Implementing a Static Route
 
-We get a routing error. Let's draw the route.
+Let's try this out in our application, I'm going to continue using the ```blog-flash``` application that I walked through in the Rails Basic lesson.
```
Comment (jmburges) [inline README.md]: As much as possible, we try and reduce mentioning previous lessons. This is mostly to reduce dependencies. Similar to code, dependecies are fine...as long as they are worth the effort required to maintain them. It's a bit annoying but unless it's suuuuuper annoying I'd go with something like "Let's work this out in an example blog app. You can find the starter code in this repo"

File: README.md [hunk 3]
```diff
@@ -1,54 +1,91 @@
...
 
-Let's draw a simple route, like /about. Let's have that map to a StaticController and an about action (though don't create that controller). Reload the page. No more routing error, now it can't find the controller.
+To begin, startup the rails server and go to ```localhost:3000/about``` as you will see this throws a routing error: ```No route matches [GET] "/about"```, to fix this, stop the rails server by pressing ```control + c``` (anytime you make a routing change you need to restart the rails server)
```
Comment (jmburges) [inline README.md]: Love going from error messages to solution!

File: README.md [hunk 4]
```diff
@@ -1,54 +1,91 @@
...
 
-Let's make a new controller, no generators, with touch app/controllers/static_controller.rb.
+Now draw the route by opening the ```config/routes.rb``` file and add the following route inside of the ```draw``` block:
 
-Controllers need to inherit from ApplicationController and explain briefly the controller naming convention.
+```ruby
+get 'about', to: 'static#about'
```
Comment (jmburges) [inline README.md]: Can you break down the three parts of this? The verb (they know HTTP verbs) the path (they know paths) and then the `static#about` They know MVC.

File: README.md [hunk 5]
```diff
@@ -1,54 +1,91 @@
...
+````
 
-Define our first controller action as a method, def about.
+Now start the rails server back up and go back to ```localhost:3000/about``` and click refresh, you should now see that the error message has changed, it's no longer complaining about not having a route, the error should now say: ```uninitialized constant StaticController```
 
-Let's hit refresh. Now it complains about a view.
+Let's fix this by creating a new controller for our static pages, you can do this manually by running the following command in the terminal:
 
-Controller actions are responsible for determining what the response is - responses are built by views.
+```touch app/controllers/static_controller.rb```
```
Comment (jmburges) [inline README.md]: I would drop this. They know how to make new files

Comment (jmburges) [inline README.md]: A lot use sublime text and such so don't want to confuse them with other commands. I've taught this before and students thought `touch` was a rais command haha

File: README.md [hunk 6]
```diff
@@ -1,54 +1,91 @@
...
+This will create a blank controller file that we can use to map to the routing file. Since there are a number of methods built into the Rails controller system, you will also want the controller to inherit from the application controller. The new file should have code that looks like this:
 
-Let's just put some basic HTML there.
+```ruby
+class StaticController < ApplicationController
+end
+```
 
-Okay, now everything works.
+The standard naming convention for controllers is the name of the controller followed by the word ```controller```.
```
Comment (jmburges) [inline README.md]: capital `C` controller

File: README.md [hunk 7]
```diff
@@ -1,54 +1,91 @@
...
 
-Let's make this rendering process of choosing a view a little more explicit using the rails render method (equiv to sinatra's erb method if you're familair with it).
+If you refresh the browser now you will see a new error: ```The action 'about' could not be found for StaticController```
```
Comment (jmburges) [inline README.md]: "This means that it found our controller (woot!) but couldn't find the action `about` in that controller"

Comment (jmburges) [inline README.md]: also comment on how for changes to controllers, you *don't* need to restart the server

File: README.md [hunk 8]
```diff
@@ -1,54 +1,91 @@
...
 
-Render can accept a string that points to a view template relative to 'views', so we can use render 'static/about'. If you think about it, that's a great place to put a view that relates to the about action in the static controller. In fact, it's such a reasonable ad convineinent place to put it that rails actually assumes, given no other render instructions, that's where you want to put it. This is a big principal in rails, convention over configuration. That's why our action worked without a render call. As a beginner we believe being explicit is easier at first then being implicit, learn the conventions and then use them.
+We're making good progress (even though we're using EDD - error driven development), and it's good to see each of the errors so that when you encounter these in your real world projects you will know how to fix them. This current error is fixed by adding the following method in the static controller:
```
Comment (jmburges) [inline README.md]: so much yes

File: README.md [hunk 9]
```diff
@@ -1,54 +1,91 @@
...
+def about
+end
+```
 
-draw a route.
-map it to a controller action.
-program the action
-create a template to render
-render that template.
+Hitting refresh in the browser will give you a 'Template is missing' error, specifically it says: ```Missing template static/about...```
```
Comment (jmburges) [inline README.md]: Another quick note that you didn't need to restart the server

File: README.md [hunk 10]
```diff
@@ -1,54 +1,91 @@
...
+
+We're very close to getting our view to show up, but before we make our final addition it's important to understand the difference between explicit and implicit rendering for the views. Rails gives us two options for how views are mapped between the controller and view files:
```
Comment (jmburges) [inline README.md]: I would switch the order of the sentences "it's important to understand ..." and "Rails give us two options...".  Much nicer to set up what you're about to talk about and give context, then give some specifics (with terminology!)

File: README.md [hunk 11]
```diff
@@ -1,54 +1,91 @@
...
+
+First let's try out explicit rendering, create a new directory in the view's directory called ```static``` and create a new file called ```some_page.html.erb```. In that file add some basic HTML code, such as:
+
+```html
+<h1>Hello from some page</h1>
+```
+
+Inside the ```about``` method in the controller add the following code:
+
+```render "static/some_page"``` you can also use: ```render "some_page"``` (Rails will automatically look inside the view directory with the same name as the controller, but you can also give the full view path).
```
Comment (jmburges) [inline README.md]: Whenever you give an option of two ways to do something. Give an opinion on which one is better

Comment (jmburges) [inline README.md]: Also just to give them a bit of context I'd just put the full code snippet.

Modify your `about` method to look like this:

```
def about
  render "static/some_page"
end
```

File: README.md [hunk 12]
```diff
@@ -1,54 +1,91 @@
...
+
+If you refresh the ```/about``` page in the browser you will see our heading of **Hello from some page**
+
+To compare that with how Rails utilizes implicit view rendering, create a new file in the static view directory called: ```about.html.erb``` and add some HTML code such as:
+
+```html
+<h1>Hello from the about page</h1>
+```
+
+Now in the controller, remove the ```render``` call completely. If you go and refresh the browser you will now see **Hello from the about page**.
```
Comment (jmburges) [inline README.md]: "So the method is empty."

Comment (jmburges) [inline README.md]: It'll be a bit crazy to the students that you can have a completely empty method that does so much. I'd mention that due to a lot of implicit assumptions, this method is doing a ton!

File: README.md [hunk 13]
```diff
@@ -1,54 +1,91 @@
...
+
+So is explicit or implicit better? Typically you will discover that you will want to utilize the implicit workflow in your day to day coding practice, the rationale is quite practical. Imagine that you are taking over a legacy Rails project, as you are getting acclimated to the code, would you prefer that the previous dev followed a standard naming process or would you rather be forced to looking through each controller code to see how the controller actions were mapped to the views? Rails has always had the goal of making the development process as efficient as possible, which is why it is typically best to follow these types of implicit procedures. With that being said, it is important to understand how the views are mapped to the controller, which is why we walked through the explicit process.
```
Comment (jmburges) [inline README.md]: great explanation

**Comment by jmburges** [comment]
Few comments!

**Comment by jordanhudgens** [comment]
@AnnJohn I implemented each of the changes requested by Joe and integrated the link from Adam

**Comment by jmburges** [comment]
one small thing then go ahead and merge it in. You should have merge rights!

**Pull request merged**